alchemist-incarnation-collektive/src/main/kotlin/it/unibo/alchemist/CollektiveIncarnation.kt

```diff
@@ -0,0 +1,97 @@
+package it.unibo.alchemist
+
+import it.unibo.alchemist.actions.RunCollektiveProgram
+import it.unibo.alchemist.model.Action
+import it.unibo.alchemist.model.Actionable
+import it.unibo.alchemist.model.Condition
+import it.unibo.alchemist.model.Context
+import it.unibo.alchemist.model.Environment
+import it.unibo.alchemist.model.Incarnation
+import it.unibo.alchemist.model.Molecule
+import it.unibo.alchemist.model.Node
+import it.unibo.alchemist.model.Position
+import it.unibo.alchemist.model.Reaction
+import it.unibo.alchemist.model.TimeDistribution
+import it.unibo.alchemist.model.conditions.AbstractCondition
+import it.unibo.alchemist.model.molecules.SimpleMolecule
+import it.unibo.alchemist.model.nodes.GenericNode
+import it.unibo.alchemist.model.reactions.Event
+import it.unibo.alchemist.model.timedistributions.DiracComb
+import it.unibo.alchemist.model.times.DoubleTime
+import it.unibo.alchemist.util.RandomGenerators.nextDouble
+import it.unibo.collektive.alchemist.device.CollektiveDevice
+import org.apache.commons.math3.random.RandomGenerator
+import org.danilopianini.util.ListSet
+
+/**
+ * Collektive incarnation in Alchemist.
+ */
+class CollektiveIncarnation<P> : Incarnation<Any?, P> where P : Position<P> {
+    override fun getProperty(node: Node<Any?>, molecule: Molecule, property: String?): Double =
+        when (val data = node.getConcentration(molecule)) {
+            is Double -> data
+            is Number -> data.toDouble()
+            is String -> data.toDoubleOrNull() ?: Double.NaN
+            else -> error("$data is not a doublify-able value")
+        }
+
+    override fun createMolecule(s: String) = SimpleMolecule(s)
+
+    override fun createConcentration(s: String?) = s
+
+    override fun createConcentration() = 0.0
```

Comment:
What does it mean this `0.0`?

Reply:
this should have been null, because it's what you create when the concentration of a molecule is empty.